garrysmod/lua/includes/modules/hook.lua

```diff
@@ -22,11 +25,11 @@ function GetTable() return Hooks end
 -----------------------------------------------------------]]
 function Add( event_name, name, func )
 
-	if ( !isfunction( func ) ) then return end
-	if ( !isstring( event_name ) ) then return end
+	if ( !isstring( event_name ) ) then ErrorNoHalt( "bad argument #1 to 'Add' (string expected, got " .. type( event_name ) .. ")" ) end
+	if ( !isfunction( func ) ) then ErrorNoHalt( "bad argument #3 to 'Add' (function expected, got " .. type( func ) .. ")" ) end

```

Comment:
If you are using `ErrorNoHalt` then you probably want the `return` still there to prevent further errors.